Generator.Equals/Models/EqualityTypeModelTransformer.cs

```diff
@@ -0,0 +1,112 @@
+﻿using System;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using Generator.Equals.Extensions;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace Generator.Equals.Models;
+
+internal sealed class EqualityTypeModelTransformer
+{
+    private readonly GeneratorAttributeSyntaxContext _context;
+    private readonly CancellationToken _token;
+
+    public EqualityTypeModelTransformer(GeneratorAttributeSyntaxContext context, CancellationToken token)
+    {
+        _context = context;
+        _token = token;
+    }
+
+    public EqualityTypeModel Transform()
+    {
+        var attributesMetadata = AttributesMetadata.Instance;
+
+        var equatableAttributeData = _context.Attributes.SingleOrDefault();
+        if (equatableAttributeData is null || !attributesMetadata.Equatable.Equals(equatableAttributeData.AttributeClass))
+        {
+            throw new Exception("Expected exactly one EquatableAttribute.");
+        }
+
+
+        var explicitMode = equatableAttributeData.GetNamedArgumentValue("Explicit") is true;
+        var ignoreInheritedMembers = equatableAttributeData.GetNamedArgumentValue("IgnoreInheritedMembers") is true;
+
+        if (_context.TargetSymbol is not ITypeSymbol symbol)
+        {
+            throw new Exception("Expected a type symbol.");
+        }
+
+        var baseTypeName = symbol.BaseType?.ToFQF();
+
+        // Extract the symbol's fully qualified name
```

Comment:
What does this comment mean, especially when right on top of something that is _not_ the fully qualified name?

Reply:
Little whoopsie dasy 😅 Moved the lines and forgot to take the comment with it